src/orchestrator/chat_completions_detection.rs

```diff
@@ -147,31 +173,129 @@ impl Orchestrator {
                         .unwrap()
                         .as_secs() as i64,
                     detections: Some(ChatDetections {
-                        input: input_detections,
+                        input: detections,
                         output: vec![],
                     }),
                     warnings: vec![OrchestratorWarning::new(
-                        InputWarningReason::UnsuitableInput,
+                        DetectionWarningReason::UnsuitableInput,
                         UNSUITABLE_INPUT_MESSAGE,
                     )],
                     ..Default::default()
                 })))
             } else {
-                let model_id = task.request.model.clone();
                 let client = ctx
                     .clients
                     .get_as::<OpenAiClient>("chat_generation")
                     .expect("chat_generation client not found");
                 let mut chat_request = task.request;
+                let model_id = chat_request.model.clone();
                 // Remove detectors as chat completion server would reject extra parameter
                 chat_request.detectors = None;
-                client
-                    .chat_completions(chat_request, task.headers)
+                let headers = task.headers.clone();
+                let chat_completions = client
+                    .chat_completions(chat_request, headers)
                     .await
                     .map_err(|error| Error::ChatGenerateRequestFailed {
-                        id: model_id,
+                        id: model_id.clone(),
                         error,
-                    })
+                    })?;
+
+                if let ChatCompletionsResponse::Unary(ref chat_completion) = chat_completions {
+                    let choices = Vec::<ChatMessageInternal>::from(chat_completion);
+
+                    let output_detections = match detectors.output {
```

Comment:
nit: since this function has gotten quite big, can we split out all of this logic (fetching the output detection part) into separate function ?

Reply:
Sounds good. I will break out the output detections into its own function.